### PR TITLE
Growth/minimalist tabs

### PR DESCRIPTION
**Pull request by ap-nri**
Please use the template that's relevant for your situation and delete the other templates. If this is just a noredink-ui repo doc change, you don't need to follow a template.

# :star2: Adding a new component

## About the component

We are adding a minimal Tab component with an understated view and NO CUSTOMIZATION if you want customizations go to Tabs please and thank you

This is to support the redesign of the teacher dashboard, specifically the tabbed interface of the class cards separating assignments by their Grading/Last Due/Up Next status as seen in the below design from figma
image

[full figma design](https://www.figma.com/file/CRqP1UX0J8lXH4GlHzu0mR/Growth-(2023%2B)?type=design&node-id=342-2383&t=FYSOdxPhU4W5MLvh-0) cc @NoRedInk/design 

### normal

image

### highlighting spacing

image

## zoomed way in to show that text ain't move when you change tabs

https://github.com/NoRedInk/noredink-ui/assets/75996561/7d7525c1-938a-45bb-b334-ea20b8323717

https://github.com/NoRedInk/noredink-ui/assets/75996561/17c086b3-881c-4220-974f-2ff11a0383f4

## Component completion checklist

- [x] I've gone through the relevant sections of the [Development Accessibility guide](https://paper.dropbox.com/doc/Accessibility-guide-4-Development--BiIVdijSaoijjOuhz3iTCJJ1Ag-rGoHpC91pFg3zTrYpvOCQ) with this component in mind
- [x] Component has clear documentation
- [x] Component is in the Component Catalog
    - [x] Component is categorized reasonably (see [Category](https://github.com/NoRedInk/noredink-ui/blob/master/component-catalog-app/Category.elm) for all the currently available categories). The component can be in multiple categories, if appropriate.
    - [x] Component has a representative preview for the Component Catalog cards (bonus points for making it delightful!)
    - [x] Component has a customizable example. Aim for having _every_ possible supported version of the component displayable through the configuration on this page. (Protip: This is handy for testing expected behavior!)
      - It helps that it is not in fact customizable.
    - [x] The customizable example produces sample code
    - [x] The component's example page includes keyboard behavior, if any
    - [x] The component's example page includes guidance around how to use the component, if necessary
- [x] Component is tested
    - axe checks and percy snapshots will be automatically added for every component. However, if you want to customize _when_ the checks and snapshots are made, you will need to make changes to [script/puppeteer-tests.js](https://github.com/NoRedInk/noredink-ui/blob/master/script/puppeteer-tests.js).
    - there are 2 ways to add Elm tests:
        - if you want to test the Component Catalog example directly, add ProgramTest-style tests under `component-catalog/tests`
        - if you want to test the component directly, add tests under `tests/Spec`. Historically, this has been the more popular Elm testing strategy for noredink-ui.
- [ ] Component API follows standard patterns in noredink-ui
    - e.g., as a dev, I can conveniently add an `nriDescription`
      - This is not supported. But it follows all the conventions of Tabs.V8 which also does not support custom wrapper attributes so if it's cool there is it cool here? Is it actually supported?
    - and adding a new feature to the component will _not_ require major API changes to the component
      - The main idea here is that once you're adding features you should be using the full Tab component. But yeah it's very possible to add configuration to this without breaking anything!
- [x] Please assign [team-accessibilibats-a11ybats](https://github.com/orgs/NoRedInk/teams/team-accessibilibats-a11ybats) as a reviewer in addition to assigning a reviewer from your team

### CODE REVIEW

File: tests/Spec/Nri/Ui/MinimalTabs.elm
Comment (tesk9) [inline tests/Spec/Nri/Ui/MinimalTabs.elm:5]: Could we update to tests MinimalTabs, please?

File: tests/Spec/Nri/Ui/MinimalTabs.elm
Comment (tesk9) [inline tests/Spec/Nri/Ui/MinimalTabs.elm:14]: This line will need to change too.

**Review by tesk9 — COMMENTED** [review]
I just looked at the tests, because I saw you didn't check off the tests in the checklist. Will look at the rest separately!

**Review by tesk9 — COMMENTED** [review]
Sorry, but I'm giving you a very piece-meal review! 😅 

I have some suggestions for the example page here: https://github.com/NoRedInk/noredink-ui/pull/1414

Additionally, I want to call out something _super_ persnickety. Because we're adding border-bottom to the selected tab, the vertical text alignment of the tab is changing ever so slightly on selection:

https://github.com/NoRedInk/noredink-ui/assets/8811312/ce70292d-4268-43b7-8c8b-d00170873940

I don't think this is desired (although in Ravi's absence, maybe @bendansby can be the authority on this).

I also think the font-weight is supposed to be 700/bold, although again maybe @bendansby can be the design approver on this.

**Comment by bendansby** [comment]
Gotta pull the ol' "what Tessa said" on both items!

**Comment by ap-nri** [comment]
> Sorry, but I'm giving you a very piece-meal review! 😅
> 
> I have some suggestions for the example page here: #1414

merged and I'm going to do the romance adjective order suggested by Ben. 

> Additionally, I want to call out something _super_ persnickety. Because we're adding border-bottom to the selected tab, the vertical text alignment of the tab is changing ever so slightly on selection:

This is me messing up the implementations. It's not supposed to be two borders. Will fix. 

> I also think the font-weight is supposed to be 700/bold, although again maybe @bendansby can be the design approver on this.

Ten four good buddy.

**Comment by ap-nri** [comment]
added something with `display:block;` above the example in the example to highlight default spacing above for @bendansby

**Comment by tesk9** [comment]
Oh! I thought the padding was supposed to be "above the tab panel", meaning in between the tablist and the tab panel?

**Comment by bendansby** [comment]
Yes, exactly. Spacing here so content is always the same distance away from the tabs

Napkin 16 06-23-23, 11 11 12 AM

**Comment by ap-nri** [comment]
I prostrate myself at the altar of your forgiveness @tesk9 @bendansby

**Comment by tesk9** [comment]
Will review on Monday!

**Review by tesk9 — COMMENTED** [review]
Looks good overall!

Because borders (including transparent colors) are co-oerced in high-contrast mode, it's difficult to tell which tab is selected in high-contrast mode. To my eye, the default blue and white colors are difficult to distinguish.

Screen Shot 2023-06-26 at 9 57 13 AM

I'll let @bendansby make a call on whether we should consider this blocking or not (since the tabs are only planned to be used teacher-side at the moment and it's a visual thing).

**Review by bendansby — APPROVED** [review]
Yeah I think we'll hold off on treating HC mode support for teachers as blockers for now 👍

**Review by tesk9 — APPROVED** [review]

**Pull request merged**
